pkg/analyzer/detection/detection.go

```diff
@@ -21,9 +22,12 @@ const LogLayerName = "layer"
 // layers, or the root layer, use `nil`. Notice that language components are not
 // extracted by DetectComponents, but if provided they are annotated with
 // certified RHEL dependencies, and returned.
-func DetectComponents(name string, files analyzer.Files, parent *database.Layer, languageComponents []*component.Component, uncertifiedRHEL bool) (*database.Namespace, []database.FeatureVersion, *database.RHELv2Components, []*component.Component, error) {
+func DetectComponents(name string, files analyzer.Files, parent *database.Layer, languageComponents []*component.Component, uncertifiedRHEL bool, isRHCOSRequired bool) (*database.Namespace, []database.FeatureVersion, *database.RHELv2Components, []*component.Component, error) {
```

Comment:
I think it may be better to have something like `DetectComponentOpts` which has attributes `uncertifiedRHEL` and maybe a slice of supported namespaces. This way, we don't have to keep adding parameters to the function. It also makes it more explicit what the parameters mean